### PR TITLE
[Gluten-229] Support Spark SortShuffleManager for ClickHouse Backend

### LINKED ISSUE #229: Support Spark SortShuffleManager for ClickHouse Backend
**Is your feature request related to a problem or challenge? Please describe what you are trying to do.**
(This section helps developers understand the context and *why* for this feature, in addition to  the *what*)

Support columnar shuffle thought Spark SortShuffleManager for Gluten + ClickHouse Backend .

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by zzcclp**
This pr includes :

1. Support using SortShuffleManager to execute columnar shuffle
2. Fix some issues after rebasing main branch

Close #229 .

## What changes were proposed in this pull request?

(Please fill in changes proposed in this fix)

## How was this patch tested?

(Please explain how this patch was tested. E.g. unit tests, integration tests, manual tests)

(If this patch involves UI changes, please attach a screenshot; otherwise, remove this)

### CODE REVIEW

**Comment by rui-mo** [comment]
LGTM. Verified on Jenkins.

File: jvm/src/main/scala/io/glutenproject/extension/StrategyOverrides.scala [hunk 1]
```diff
@@ -76,8 +69,26 @@ object JoinSelectionOverrides extends Strategy with JoinSelectionHelper with SQL
       if (GlutenConfig.getSessionConf.forceShuffledHashJoin) {
         // Force use of ShuffledHashJoin in preference to SortMergeJoin. With no respect to
         // conf setting "spark.sql.join.preferSortMergeJoin".
-        val leftBuildable = canBuildShuffledHashJoinLeft(joinType)
-        val rightBuildable = canBuildShuffledHashJoinRight(joinType)
+        val (leftBuildable, rightBuildable) = if (GlutenConfig.getConf.isClickHouseBackend) {
+          // Currently, ClickHouse backend can not support AQE, so it needs to use join hint
+          // to decide the build side, after supporting AQE, will remove this.
```
Comment (zml1206) [inline jvm/src/main/scala/io/glutenproject/extension/StrategyOverrides.scala:74]: What is the relationship between do not support AQE and needs to use join hint? I'm preparing to move JoinSelectionOverrides to ColumnarOverrides. Can you help explain why? @zzcclp 
Thank you.

Comment (zml1206) [reply jvm/src/main/scala/io/glutenproject/extension/StrategyOverrides.scala:74]: cc @zhztheplayer

Comment (zzcclp) [reply jvm/src/main/scala/io/glutenproject/extension/StrategyOverrides.scala:74]: remove these comments please, it's not correct, now it already supports AQE.

Comment (zml1206) [reply jvm/src/main/scala/io/glutenproject/extension/StrategyOverrides.scala:74]: OK.

**Pull request merged**